### PR TITLE
Add SPV_KHR_untyped_pointers

### PR DESCRIPTION
**Pull request by alan-baker**
Provisionally ratified

### CODE REVIEW

**Comment by dneto0** [comment]
The branch has conflicts.

**Review by Naghasan — APPROVED** [review]

**Review by dneto0 — APPROVED** [review]

File: extensions/KHR/SPV_KHR_untyped_pointers.asciidoc [hunk 1]
```diff
@@ -0,0 +1,1211 @@
...
+ +
+'Result Type' must be an *OpTypeUntypedPointerKHR*. +
+ +
+'Storage Class' is the Storage Class of the memory holding the object. It must not be *Generic*. It must be the same storage class as the 'Storage Class' operand of the 'Result Type'. +
+ +
+'Data Type' is optional. It is the type of the object in memory. 'Data Type' must be specified if 'Storage Class' is *Function*, *Private*, or *Workgroup*. Refer to the client API for other storage classes. +
+ +
+'Initializer' is optional. If 'Initializer' is present, it will be the initial value of the variable’s memory content. 'Initializer' must be an '<id>' from a constant instruction or a global (module scope) <<Variable,variable>>. 'Initializer' must have the same type as 'Data Type'. 2+| Capability: +
+*UntypedPointersKHR*
+| 5 + variable | 4418 | '<id> Result Type' | 'Result <id>' | 'Storage Class' | Optional '<id> Data Type' | Optional '<id> Initializer'
```
Comment (Naghasan) [inline extensions/KHR/SPV_KHR_untyped_pointers.asciidoc:912]:
```suggestion
| 4 + variable | 4418 | '<id> Result Type' | 'Result <id>' | 'Storage Class' | Optional '<id> Data Type' | Optional '<id> Initializer'
```

File: extensions/KHR/SPV_KHR_untyped_pointers.asciidoc [hunk 2]
```diff
@@ -0,0 +1,1211 @@
...
+ +
+'Result Type' must be an *OpTypeInt* with 32-bit 'Width' and 0 'Signedness'. +
+ +
+'Structure' must be a *Block*-decorated *OpTypeStruct* whose last member is a run-time array. +
+ +
+'Pointer' must be a <<PointerType,pointer type>>. 'Pointer' must have the same value as a descriptor. That is, the value must be the same as a <<Variable,variable>> decorated with *DescriptorSet* and *Binding* or an element in such a variable when the data type is an array of *Block*-decoarated structures. +
+ +
+'Array member' is an unsigned 32-bit integer index of the last member of 'Structure'. That member’s type must be from *OpTypeRuntimeArray*. 2+| Capability: +
+*UntypedPointersKHR*
+| 7 | 4425 | '<id> Result Type' | 'Result <id>' | '<id> Structure' | '<id> Pointer' | 'Literal Array member'
```
Comment (Naghasan) [inline extensions/KHR/SPV_KHR_untyped_pointers.asciidoc:978]:
```suggestion
| 6 | 4425 | '<id> Result Type' | 'Result <id>' | '<id> Structure' | '<id> Pointer' | 'Literal Array member'
```

File: extensions/KHR/SPV_KHR_untyped_pointers.asciidoc [hunk 3]
```diff
@@ -0,0 +1,1211 @@
...
+The value must be between 0 (for no locality) and 3 (for extreme locality) inclusive. +
+ +
+'Cache Type' is optional.
+If 'Cache Type' is present, it specifies the type of cache.
+It must be a constant instruction with an integer scalar type.
+The value must be either 0 (for instruction cache) or 1 (for data cache). +
+ +
+The default values of all optional operands are implementation defined. 2+| Capability: +
+*UntypedPointersKHR*
+| 4+ | 4426 | '<id> Pointer Type' | '<id> Num Bytes' | Optional '<id> RW' | Optional '<id> Locality' | Optional '<id> Cache Type'
```
Comment (Naghasan) [inline extensions/KHR/SPV_KHR_untyped_pointers.asciidoc:1009]:
```suggestion
| 3  + variable | 4426 | '<id> Pointer Type' | '<id> Num Bytes' | Optional '<id> RW' | Optional '<id> Locality' | Optional '<id> Cache Type'
```

**Review by Naghasan — COMMENTED** [review]
Noticed the word count issues while cross-checking https://github.com/KhronosGroup/SPIRV-Headers/pull/439

**Comment by alan-baker** [comment]
> Noticed the word count issues while cross-checking [KhronosGroup/SPIRV-Headers#439](https://github.com/KhronosGroup/SPIRV-Headers/pull/439)

Thanks, fixed now.

**Comment by bashbaug** [comment]
I think the HTML file needs to be updated to be in sync with the latest asciidoc changes.

**Comment by alan-baker** [comment]
@bashbaug I think everything is updated now. PTAL

**Review by bashbaug — APPROVED** [review]
Thanks, latest version LGTM.  Merging as we discussed in the July 17th teleconference.

**Pull request merged**